pyoozie/workflow.py

```diff
@@ -0,0 +1,43 @@
+# Copyright (c) 2017 "Shopify inc." All rights reserved.
+# Use of this source code is governed by a MIT-style license that can be found in the LICENSE file.
+import typing  # pylint: disable=unused-import
+
+from pyoozie import tags  # pylint: disable=unused-import
+
+
+class AbstractAction(object):
```

Comment:
What does it mean to be an `AbstractAction`? Scanning the subclasses below, it seems that the common element in all (except for `Kill`) is that they have an error handler. That makes sense, given what we talked about yesterday.

So, what do you think about
 - making `AbstractAction` an [ABC](https://docs.python.org/2/library/abc.html),
 - adding an abstract method `on_error()` to it, and
 - removing `Kill` from the hierarchy (since it doesn't make sense to have an error handler for `Kill`)?

Since this design is meant to make error handling tractable, I think it makes sense to be explicit about error handling intentions.